src/liboslexec/oslexec_pvt.h

```diff
@@ -1750,7 +1759,8 @@ class OSLEXECPUBLIC ShadingContext {
     PerThreadInfo *m_threadinfo;        ///< Ptr to our thread's info
     mutable TextureSystem::Perthread *m_texture_thread_info; ///< Ptr to texture thread info
     ShaderGroup *m_group;               ///< Ptr to shader group
-    std::vector<char> m_heap;           ///< Heap memory
+    /// Heap memory
+    std::vector<char, pvt::aligned_allocator<char,OIIO_CACHE_LINE_SIZE>> m_heap;
```

Comment:
Are you sure this needs to be an `std::vector` ? Just having a plain `unique_ptr` would get rid of all the allocator boilerplate.

`std::vector` in this case has extra costs, like 0-init and copying on resize (which AFAICT is not important to this use case)

Reply:
As a practical matter, I don't know that those costs hurt -- this tends to get resized a couple of times until you've shaded the biggest material, then it just sits there.

But, hmmm, you raise an excellent point. Maybe I'll just rewrite it to use unique_ptr and a simple aligned malloc. Let me see if that looks cleaner.